server/src/main/java/org/opensearch/common/settings/SettingsModule.java

```diff
@@ -179,6 +179,34 @@ public void configure(Binder binder) {
         binder.bind(IndexScopedSettings.class).toInstance(indexScopedSettings);
     }
 
+    /**
+     * Dynamically registers a new Setting at Runtime. This method is mostly used by plugins/extensions
+     * to register new settings at runtime. Settings can be of Node Scope or Index Scope.
+     * @param setting which is being registered in the cluster.
+     * @return boolean value is set to true when successfully registered, else returns false
+     */
+    public boolean registerDynamicSetting(Setting<?> setting) {
+        boolean onNodeSetting = false;
+        boolean onIndexSetting = false;
+        try {
+            if (setting.hasNodeScope()) {
+                onNodeSetting = clusterSettings.registerSetting(setting);
+            }
+            if (setting.hasIndexScope()) {
+                onIndexSetting = indexScopedSettings.registerSetting(setting);
+            }
+            registerSetting(setting);
```

Comment:
```suggestion
try {
            registerSetting(setting);
            if (onNodeSetting || onIndexSetting) {
                logger.info("Registered new Setting: " + setting.getKey() + " successfully ");
                return true;
            }
} catch (final IllegalArgumentException ex) {
            if (onNodeSetting) {
                clusterSettings.unregisterSetting(setting);
            }
            
            if (onIndexSetting) {
                indexScopedSettings.unregisterSetting(setting);
            }
}
```

Needs new `unregisterSetting` to compensate possible registration failures

Reply:
I'm implementing unregisterSetting method in `AbstractScopedSettings.java` that removes the key from the respective settings maps.  Let me know if there is anything else that needs to be done in that method.

Reply:
I think it should be fine, as far as settings are consistent, thank you